src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -222,43 +315,53 @@ def _store_testcase_for_regression_testing(testcase, testcase_file_path):
                    regression_testcase_url)
 
 
-def find_fixed_range(testcase_id, job_type):
-  """Attempt to find the revision range where a testcase was fixed."""
-  deadline = tasks.get_task_completion_deadline()
+def utask_preprocess(testcase_id, job_type, uworker_env):
   testcase = data_handler.get_testcase_by_id(testcase_id)
   if not testcase:
-    return
+    return None
 
   if testcase.fixed:
     logs.log_error('Fixed range is already set as %s, skip.' % testcase.fixed)
-    return
-
-  # Setup testcase and its dependencies.
-  _, testcase_file_path, error = setup.setup_testcase(testcase, job_type)
-  if error:
-    # TODO(https://github.com/google/clusterfuzz/issues/3008): Change this when
-    # progression is migrated.
-    all_errors = uworker_handle_errors.get_all_handled_errors()
-    uworker_handle_errors.handle(error, all_errors)
-    return
+    return None
 
+  # TODO: Make sure this is always properly cleared on failure/ when done. 
   # Set a flag to indicate we are running progression task. This shows pending
   # status on testcase report page and avoid conflicting testcase updates by
   # triage cron.
-  testcase.set_metadata('progression_pending', True)
+  testcase.set_metadata('progression_pending', True, True)
+  data_handler.update_testcase_comment(testcase, data_types.TaskState.STARTED)
+  
+  return uworker_io.UworkerInput(
+      job_type=job_type,
+      testcase_id=testcase_id,
+      uworker_env=uworker_env,
+      testcase=testcase,
+  )
+
+def find_fixed_range(uworker_input):
+  """Attempt to find the revision range where a testcase was fixed.""" 
+  deadline = tasks.get_task_completion_deadline()
+  testcase_id = uworker_input.testcase_id
+  testcase = uworker_input.testcase
+  job_type = uworker_input.job_type
+ 
+  # Setup testcase and its dependencies.
+  _, testcase_file_path, error = setup.setup_testcase(testcase, job_type)
+  if error:
+    # TODO: reset progressing_pending in postrpocess in case of errors?
```

Comment:
nit: Please put a name or issue in TODOs.
See https://google.github.io/styleguide/pyguide.html#312-todo-comments